test/grep.js

```diff
@@ -56,6 +56,24 @@ test("multiple files, one doesn't exist, one doesn't match", t => {
   t.is(result.code, 2);
 });
 
+test('-r option not provided, but directory is present', t => {
+  t.truthy(fs.existsSync('test/resources'));
+  t.truthy('/test/resources/a.txt');
+  const result = shell.grep(/oogabooga/, 'test/resources', 'test/resources/a.txt');
+  t.truthy(shell.error());
+  t.is(result.stderr, 'grep: test/resources Is a directory');
+  t.is(result.code, 2);
+});
+
+test('atleast one directory or file does not exist', t => {
+  t.truthy(fs.existsSync('test/resources'));
+  t.falsy(fs.existsSync('test/resources/random.txt'));
+  const result = shell.grep('-r', /oogabooga/, 'test/resources', 'test/resources/random.txt');
+  t.truthy(shell.error());
+  t.is(result.code, 2);
+  t.is(result.stderr, 'grep: no such file or directory: test/resources/random.txt');
```

Comment:
I think we also need to check `result.stdout`. My `grep` will return results for the folders which actually do exist.